### PR TITLE
Bump  SQLAlchemy to 2.0

### PR DESCRIPTION
**Pull request by lollerfirst**
A lot of warnings for basically the same thing: use `_mapping` for dictionary interface to `Row`.

### CODE REVIEW

**Comment by lollerfirst** [comment]
mypy is a bit dumb

**Comment by callebtc** [comment]
> mypy is a bit dumb

image

File: pyproject.toml [hunk 1]
```diff
@@ -7,7 +7,7 @@ license = "MIT"
 
 [tool.poetry.dependencies]
 python = "^3.8.1"
-SQLAlchemy = {version = "1.4.52", extras = ["asyncio"]}
+SQLAlchemy = {version = "2.0.35", extras = ["asyncio"]}
```
Comment (prusnak) [inline pyproject.toml:10]:
```suggestion
SQLAlchemy = {version = "^2.0.35", extras = ["asyncio"]}
```

**Review by prusnak — CHANGES_REQUESTED** [review]

File: poetry.lock [hunk 2]
```diff
@@ -2059,20 +2156,24 @@ files = [
...
-test = ["big-O", "importlib-resources", "jaraco.functools", "jaraco.itertools", "jaraco.test", "more-itertools", "pytest (>=6,!=8.1.*)", "pytest-checkdocs (>=2.4)", "pytest-cov", "pytest-enabler (>=2.2)", "pytest-ignore-flaky", "pytest-mypy", "pytest-ruff (>=0.2.1)"]
+enabler = ["pytest-enabler (>=2.2)"]
+test = ["big-O", "importlib-resources", "jaraco.functools", "jaraco.itertools", "jaraco.test", "more-itertools", "pytest (>=6,!=8.1.*)", "pytest-ignore-flaky"]
+type = ["pytest-mypy"]
 
 [metadata]
 lock-version = "2.0"
 python-versions = "^3.8.1"
-content-hash = "38db19797cb04d1f941f6f58c56c8dfe09ac34d45ac2f654bc97dd8110912da8"
+content-hash = "1676da5d07d195f219cdb1338bdf693da20b971d904cae87812a30ddc31a76c0"
```
Comment (prusnak) [inline poetry.lock:2179]: on my machine running `poetry lock --no-update` leads to a different content-hash:

```suggestion
content-hash = "47f05f6093dbe0e04bb135e76c8e51cb306dd358fa5259fbbad6e24b005750d3"
```

Comment (lollerfirst) [reply poetry.lock]: should be fine now.

**Review by prusnak — APPROVED** [review]

File: cashu/mint/crud.py [hunk 3]
```diff
@@ -677,7 +677,10 @@ async def get_balance(
             """
         )
         assert row, "Balance not found"
-        return int(row[0])
+
+        # Thank SQLAlchemy for this!
+        key = next(iter(row))
```
Comment (callebtc) [inline cashu/mint/crud.py:682]:
```suggestion
        # sqlalchemy index of first element
        key = next(iter(row))
```

**Review by callebtc — APPROVED** [review]

**Pull request merged**